lib/bloc_record/selection.rb

```diff
@@ -1,17 +1,250 @@
 require 'sqlite3'
 
 module Selection
-	def find(id)
-		# straight forward SELECT that returns the row matching the id passed in
-		# why we are using columns instead of (*)?
+	def get_min
+		min = connection.execute <<-SQL
+			SELECT id FROM #{table}
+			ORDER BY id ASC LIMIT 1;
+		SQL
+		min[0][0]
+	end
+
+	def get_max
+		max = connection.execute <<-SQL
+			SELECT id FROM #{table}
+			ORDER BY id DESC LIMIT 1;
+		SQL
+		max[0][0]
+	end
```

Comment:
IDs are 
1. >= id.first
2. <= id.last or similarly <= .count
3. Integers

All of the IDs ought to be in AddressBloc in memory when the program initializes so I can cross-reference the IDs against that list first. Would that be 

I added this to models/address_book.rb 

```ruby
  def find_entry_by_id(id)
    Entry.where(id: id, address_book_id: self.id).first
  end
```

I suppose I can call `@address_book.entries.find_entry_by_id(id)`  within the controller to boot out deleted ids. I am unsure how to access the instance of AddressBook from the BlocRecord side, however, to check for valid IDs, or if I want to.